webrender/src/tiling.rs

```diff
@@ -702,6 +704,23 @@ impl RenderTask {
 pub const SCREEN_TILE_SIZE: i32 = 64;
 pub const RENDERABLE_CACHE_SIZE: DevicePixel = DevicePixel(2048);
 
+/// Per-batch clipping info merged with the mask image.
+#[derive(Clone, Debug)]
+pub struct MaskedClip {
```

Comment:
Instead of introducing a new structure here, could we just add the fields/struct to the Clip structure? The Clip structure already handles simple / complex clips, it seems like it's probably reasonable to make the Clip structure able to handle any number of clip types. If we do that, we won't need to modify the API of all the FrameBuilder methods.